maven-plugin/src/main/scala/akka/grpc/maven/GenerateMojo.scala

```diff
@@ -1,22 +1,60 @@
 package akka.grpc.maven
 
-import java.io.File
+import java.io.{ ByteArrayOutputStream, File, PrintStream }
 
+import akka.grpc.gen.CodeGenerator
 import akka.grpc.gen.javadsl.{ JavaBothCodeGenerator, JavaClientCodeGenerator, JavaServerCodeGenerator }
 import akka.grpc.gen.scaladsl.{ ScalaBothCodeGenerator, ScalaClientCodeGenerator, ScalaServerCodeGenerator }
-import akka.grpc.gen.CodeGenerator
 import javax.inject.Inject
 import org.apache.maven.plugin.AbstractMojo
-import org.slf4j.LoggerFactory
-import protocbridge.{ JvmGenerator, Target }
 import org.apache.maven.project.MavenProject
+import org.sonatype.plexus.build.incremental.BuildContext
+import protocbridge.{ JvmGenerator, Target }
 import scalapb.ScalaPbCodeGenerator
 
 import scala.annotation.tailrec
 import scala.beans.BeanProperty
+import scala.util.control.NoStackTrace
 
-class GenerateMojo @Inject() (project: MavenProject) extends AbstractMojo {
-  val log = LoggerFactory.getLogger(classOf[GenerateMojo])
+object GenerateMojo {
+  val protocVersion = "-v351"
+
+  case class ProtocError(file: String, line: Int, pos: Int, message: String)
+  private val ProtocErrorRegex = """(\w+\.\w+):(\d+):(\d+):\s(.*)""".r
+
+  /** @return a left(parsed error) or a right(original error string) if it cannot be parsed */
+  def parseError(errorLine: String): Either[ProtocError, String] = {
+    errorLine match {
+      case ProtocErrorRegex(file, line, pos, message) =>
+        Left(ProtocError(file, line.toInt, pos.toInt, message))
+      case unknown =>
+        Right(unknown)
+    }
+  }
+
+  private def captureStdOutAnderr[T](block: => T): (String, String, T) = {
```

Comment:
Nice, though unfortunate that it's needed. Added https://github.com/os72/protoc-jar/issues/48

Reply:
Thanks